main.go

```diff
@@ -225,6 +235,16 @@ func main() {
 	}
 }
 
+func initializeProtocol(stdin io.Reader, stdout io.Writer) {
```

Comment:
Nit: can we rename `stdin` -> `in` and `stdout` -> `out` since they don't need to be standard in and standard out, but any readers/writers?